nomos-services/tx-service/src/tx/service.rs

```diff
@@ -526,19 +531,37 @@ where
         }
     }
 
-    async fn handle_get_transactions_message(
+    async fn partition_transactions_by_availability(
         pool: &Pool,
-        hashes: BTreeSet<Pool::Key>,
-        reply_channel: oneshot::Sender<Pin<Box<dyn futures::Stream<Item = Pool::Item> + Send>>>,
-    ) {
-        let transactions = pool
-            .get_items_by_keys(hashes)
+        hashes: Vec<Pool::Key>,
+    ) -> Result<crate::TransactionsByHashesResponse<Pool::Item, Pool::Key>, MempoolError> {
+        let keys_set: BTreeSet<Pool::Key> = hashes.into_iter().collect();
+
+        let items_stream = pool
+            .get_items_by_keys(keys_set.iter().cloned())
             .await
-            .unwrap_or_else(|_| Box::pin(futures::stream::iter(Vec::new())));
+            .map_err(|e| {
+                MempoolError::StorageError(format!("Failed to get items by keys: {e:?}"))
+            })?;
+
+        let found_transactions: Vec<Pool::Item> = items_stream.collect().await;
 
-        if let Err(_e) = reply_channel.send(transactions) {
-            tracing::debug!("Failed to send transactions reply");
+        if found_transactions.len() == keys_set.len() {
+            return Ok(crate::TransactionsByHashesResponse::new(
```

Comment:
why the fully qualified path on `TransactionByHashesResponse`? is there any similar named struct?